Sources/MessagingInApp/MessagingInAppImplementation.swift

```diff
@@ -73,12 +73,7 @@ extension MessagingInAppImplementation: GistDelegate {
         eventListener?.messageShown(message: InAppMessage(gistMessage: message))
 
         if let deliveryId = getDeliveryId(from: message) {
-            // the state of the SDK does not change if adding this queue task isn't successful so ignore result
-            // FIXME: [CDP] Pass to Journey
-            // _ = queue.addTrackInAppDeliveryTask(deliveryId: deliveryId, event: .opened)
-
-            // TODO: add the TrackInAppMetricEvent
-//            busEventManager.send(TrackInAppMetricEvent(deliveryID: deliveryId, event: "")
+            busEventManager.postEvent(TrackInAppMetricEvent(deliveryID: deliveryId, event: InAppMetric.opened.rawValue))
```

Comment:
```suggestion
            busEventManager.postEvent(TrackInAppMetricEvent(deliveryID: deliveryId, event: .opened))
```
 
Could we make the `TrackInAppMetricEvent.event` parameter be a `InAppMetric` data type? It seems easier to use that way.

Reply:
We can, i wanted to keep these `events` as raw as possible because the notification center stores them as `key value` and I want parsing and storage to be as simple as possible.